File: GenerateGBFSDataset/src/main/scala/com/david/GenerateGBFS/json/StationInformationJSON.scala
package com.david.GenerateGBFS.json

import com.david.GenerateGBFS.model.StationInformation
import org.json.JSONObject
import org.json4s.JsonDSL._
import org.json4s.jackson.JsonMethods._

class StationInformationJSON(station_id: String,
                             external_id: String,
                             name: String,
                             short_name: String,
                             lat: String,
                             lon: String,
                             rental_methods: List[String],
                             capacity: Int,
                             eightd_has_key_dispenser: Boolean,
                             has_kiosk: Boolean) extends JSON[StationInformation] {


  override def toJSON: JSONObject = {
    val json = ("station_id" -> station_id) ~
      ("external_id" -> external_id) ~
      ("name" -> name) ~
      ("short_name" -> short_name) ~
      ("lat" -> lat) ~
      ("lon" -> lon) ~
      ("rental_methods" -> rental_methods)
      ("capacity" -> capacity) ~
      ("eightd_has_key_dispenser" -> eightd_has_key_dispenser) ~
      ("has_kiosk" -> has_kiosk)

    val compactJson = compact((render(json)))

    return new JSONObject(compactJson)
  }

  override def toModel: StationInformation = {
    return new StationInformation(station_id, external_id, name, short_name, lat, lon, rental_methods, capacity, eightd_has_key_dispenser, has_kiosk)
  }
}

File: GBFSCron/src/main/java/com/david/GBFSCron/json/StationInformationJSON.java
package com.david.GBFSCron.json;

import com.david.GBFSCron.model.StationInformation;
import com.david.GBFSCron.utils.JSONParser;
import org.json.JSONObject;

import java.util.List;

public class StationInformationJSON {
    public String station_id;
    public String external_id;
    public String name;
    public String short_name;
    public String lat;
    public String lon;
    public List<String> rental_methods;
    public Integer capacity;
    public Boolean eightd_has_key_dispenser;
    public Boolean has_kiosk;

    public JSONObject toJSON(){
        return JSONParser.toJSON(this.getClass(), this);
    }

    public StationInformation toModel(){
        StationInformation stationInformation = new StationInformation();
        stationInformation.setStationId(station_id);
        stationInformation.setExternalId(external_id);
        stationInformation.setName(name);
        stationInformation.setShortName(short_name);
        stationInformation.setLat(lat);
        stationInformation.setLon(lon);
        stationInformation.setRentalMethods(rental_methods);
        stationInformation.setCapacity(capacity);
        stationInformation.setEightdHasKeyDispenser(eightd_has_key_dispenser);
        stationInformation.setHasKiosk(has_kiosk);

        return stationInformation;
    }
}
